samples/nrf9160/aws_iot/src/main.c

```diff
@@ -493,15 +501,30 @@ void main(void)
 {
 	int err;
 
-	LOG_INF("The AWS IoT sample started, version: %s", CONFIG_APP_VERSION);
+	LOG_INF("The AWS IoT sample started, version: %s", CONFIG_AWS_IOT_SAMPLE_APP_VERSION);
 
 	cJSON_Init();
 
 #if defined(CONFIG_NRF_MODEM_LIB)
 	nrf_modem_lib_dfu_handler();
 #endif
 
+#if IS_ENABLED(CONFIG_AWS_IOT_SAMPLE_DEVICE_ID_USE_HW_ID)
+	static char device_id[HW_ID_LEN];
+
+	err = hw_id_get(device_id, HW_ID_LEN);
+	if (err) {
+		LOG_ERR("Failed to retrieve device ID, error: %d", err);
+	}
+	static struct aws_iot_config config;
```

Comment:
```suggestion
	static struct aws_iot_config config = {
	    .client_id = device_id,
	    .client_id_len = strlen(device_id)
	};
```

Reply:
The values can be set directly, I would also move the declaration up next to `static char device_id[HW_ID_LEN];
And, we should use `strlen` to get the length of the string contained in `device_id` buffer. HW_ID gives the whole buffer which will usually be larger and include the null character. Including the null character is not necessary. We only want the string.